setupext.py

```diff
@@ -145,6 +161,47 @@ def check_for_openmp():
         return [], []
 
 
+def check_CPP14_flag(compile_flags):
+    # Create a temporary directory
+    ccompiler = new_compiler()
+    customize_compiler(ccompiler)
+
+    tmp_dir = tempfile.mkdtemp()
+    start_dir = os.path.abspath(".")
+
+    ok = True
+    try:
+        os.chdir(tmp_dir)
```

Comment:
Two reasons:
- os.chdir should not produce any error that's caught in this try block
- removing the return variable (see following suggestion)
```suggestion
    os.chdir(tmp_dir)
    try:
```